commonknowledge/meep-intelligence-hub
# Generalise contentEditable mutations, accounting for enter-key and de…

PR that adds a bit of table stakes UX in the process and refactors code in the process.

## Description

- Keying enter 'saves' the field
- Saving an empty field defaults to a default value

## Motivation and Context

I'd like to use this pattern more, so let's trim code and make it robust at the same time.

## How Can It Be Tested?

- Open a report, and then click-to-edit the report's title in various ways and it should be intuitive. Feel free to report any edge-cases.
- Similarly, data sources (`/inspect/...`)

## How Will This Be Deployed?

Standard.

## Types of changes
